src/controllers/supplier_controller.ts

```diff
@@ -1,74 +1,104 @@
-    try{
-        const supplierObeject: SupplierInterface = req.body;
-        validateSupplier(supplierObeject);
-        const supplier = new Supplier(req.body);
-        const savedSupplier = await supplier.save();
-        res.status(201).send(savedSupplier);
-    }catch(err){
-        res.status(400).send(err);
-    }
-}
-
-
-export const removeSupplier = async (req:Request, res:Response) :Promise<void> =>{
-    try{
-        const supplier = await Supplier.deleteMany({id: Number(req.params.id)},{new:true});
-        res.status(200).send("Supplier deleted");
-    }catch(err){
-        res.status(400).send(err);
-    }
-}
+export const addSupplier = async (
+  req: Request,
+  res: Response
+): Promise<void> => {
+  try {
+    const supplierObeject: SupplierInterface = req.body;
+    validateSupplier(supplierObeject);
+    const supplier = new Supplier(req.body);
+    const savedSupplier = await supplier.save();
+    res.status(201).send(savedSupplier);
+  } catch (err) {
+    console.log(err);
+    const status = err.statusCode() || 500;
+    const message = err.getMessage();
+    res.status(status).send(message);
+  }
+};
 
-export const updateSupplier = async (req:Request, res:Response) :Promise<void> =>{
+export const removeSupplier = async (
+  req: Request,
+  res: Response
+): Promise<void> => {
+  try {
+    await Supplier.deleteMany({ id: Number(req.params.id) }, { new: true });
+    res.status(200).send("Supplier deleted");
+  } catch (err) {
+    console.log(err);
+    const status = err.statusCode() || 500;
+    const message = err.getMessage();
+    res.status(status).send(message);
+  }
+};
 
-    try{
-        const supplierObeject: SupplierInterface = req.body;
-        validateSupplier(supplierObeject);
-        const supplier = await Supplier.findOneAndUpdate({id: Number(req.params.id)},req.body,{new:true})
-        if(supplier){
-            const prod  = await Product.updateMany({"supplier.id":Number(req.body['id'])},{$set:{"supplier.name":req.body['name']}})
-            res.status(200).send(prod)
-        }else{
-            res.status(404).send("No category");
-        }
-    }catch(err){
-        res.status(400).send(err);
+export const updateSupplier = async (
+  req: Request,
+  res: Response
+): Promise<void> => {
+  try {
+    const supplierObeject: SupplierInterface = req.body;
+    validateSupplier(supplierObeject);
+    const supplier = await Supplier.findOneAndUpdate(
+      { id: Number(req.params.id) },
+      req.body,
+      { new: true }
+    );
+    if (supplier) {
+      const prod = await Product.updateMany(
+        { "supplier.id": Number(req.body["id"]) },
+        { $set: { "supplier.name": req.body["name"] } }
+      );
+      res.status(200).send(prod);
+    } else {
+      res.status(404).send("No category");
     }
-}
\ No newline at end of file
+  } catch (err) {
+    console.log(err);
+    const status = err.statusCode() || 500;
+    const message = err.getMessage();
```

Comment:
Since you need the try catch everywhere create 2 middlewares. The first one is just a simple try catch and the second one is this try catch with the custom error and status (like the one above). Add the 2 middlewares to all functions accordingly